algo-core/src/process_input.py

```diff
@@ -0,0 +1,276 @@
+"""
+Copyright 2021 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import datetime
+import math
+import pandas as pd
+
+
+def tracks_hours_to_slots(tracks):
+    for track in tracks:
+        track["start_slot"] = track["start_hour"] * 2
+        track["end_slot"] = track["end_hour"] * 2
+    return tracks
+
+
+def get_total_slots_covered(tracks):
+    total_slots_covered = 0
+    for t, track in enumerate(tracks):
+        for d in range(track["start_day"], track["end_day"] + 1):
+            total_slots_covered += 2 * (
+                track["end_hour"] - track["start_hour"]
+            )
+    return total_slots_covered
+
+
+def get_datetime_days(start_date, num_days):
+    delta = datetime.timedelta(days=1)
+    days = [start_date]
+
+    for d in range(1, num_days):
+        days.append(days[d - 1] + delta)
+    return days
+
+
+def slots_to_range(available_slots, end_hour, allowed_availabilities):
```

Comment:
Would it make more sense to change `end_hour` to `end_slot`?  This would also match how the function was used:
```
slot_ranges = slots_to_range(
            available_slots,
            config["end_slot"],
            config["allowed_availabilities"],
        )
```

Reply:
Indeed, I've changed it to `end_slot`.